backend/src/index.spec.ts

```diff
@@ -1,28 +1,52 @@
 import { resolvers } from './resolvers'
 import { typeDefs } from './typeDefs'
+import { permissions } from './permissions'
 import { PostsDataSource } from './postsDataSource'
 import { ApolloServer } from 'apollo-server'
 import { createTestClient } from 'apollo-server-testing'
+import { applyMiddleware } from 'graphql-middleware'
+import { makeExecutableSchema } from 'graphql-tools'
+const bcrypt = require('bcrypt')
+const jwt = require('jsonwebtoken')
+require('dotenv').config()
+
+const testContext = (testToken?) => {
+  let token = testToken || ''
+  token = token.replace('Bearer ', '')
+  try {
+    const decodedJwt = jwt.verify(
+      token,
+      process.env.JWTSECRET
+    )
+    return { decodedJwt }
+  } catch (e) {
+    return {}
+  }
+}
+const setupServer = (postData, userData, testToken?) => {
+  const schema = makeExecutableSchema({ typeDefs, resolvers })
+  const server = new ApolloServer({
+    schema: applyMiddleware(schema, permissions),
+    context: testContext(testToken), // not sure if this is the correct way. but we didn`t find another solution to add the token as request (see https://github.com/apollographql/apollo-server/issues/2277)
+    dataSources: () => ({
+      posts: new PostsDataSource(postData, userData)
```

Comment:
```suggestion
      posts
```

Declare `let posts` and re-initialize `posts = new PostsDataSource(postData, userData)` in a `beforeEach`. Your code is initializing a new `PostsDataSource` *on every request* which is probably not what you want.